packages/oslo-converter-uml-ea/lib/converter-handlers/ConnectorConverterHandler.ts

```diff
@@ -1,153 +1,233 @@
-import { injectable } from 'inversify';
+import { inject, injectable } from 'inversify';
 import { TagNames } from '../enums/TagNames';
 import { ConverterHandler } from '../interfaces/ConverterHandler';
 import type { UriRegistry } from '../UriRegistry';
-import { convertToCase, getTagValue, ignore } from '../utils/utils';
+import { getTagValue, ignore } from '../utils/utils';
+import { ConnectorNormalisationService } from '../ConnectorNormalisationService';
+import { EaUmlConverterServiceIdentifier } from '../config/EaUmlConverterServiceIdentifier';
 
 @injectable()
 export class ConnectorConverterHandler extends ConverterHandler<NormalizedConnector> {
-  public async filterIgnoredObjects(model: DataRegistry): Promise<DataRegistry> {
-    model.connectors = model.connectors.filter(x => !ignore(x));
+  @inject(EaUmlConverterServiceIdentifier.ConnectorNormalisationService)
+  private readonly connectorNormalisationService!: ConnectorNormalisationService;
+
+  public async filterIgnoredObjects(
+    model: DataRegistry
+  ): Promise<DataRegistry> {
+    model.connectors = model.connectors.filter((x) => !ignore(x));
 
     return model;
   }
 
-  public async convert(model: DataRegistry, uriRegistry: UriRegistry, store: QuadStore): Promise<QuadStore> {
+  public async convert(
+    model: DataRegistry,
+    uriRegistry: UriRegistry,
+    store: QuadStore
+  ): Promise<QuadStore> {
     model.normalizedConnectors
-      .filter(x => model.targetDiagram.connectorsIds.includes(x.originalId))
-      .forEach(object => store.addQuads(this.createQuads(object, uriRegistry, model)));
+      .filter((x) => model.targetDiagram.connectorsIds.includes(x.originalId))
+      .forEach((object) =>
+        store.addQuads(this.createQuads(object, uriRegistry, model))
+      );
 
     return store;
   }
 
   public async normalize(model: DataRegistry): Promise<DataRegistry> {
-    model.normalizedConnectors = model.connectors
-      .filter(x => model.targetDiagram.connectorsIds.includes(x.id))
-      .flatMap(x => this.normalizeConnector(x, model.elements));
+    const tasks: Promise<NormalizedConnector[]>[] = [];
+    model.connectors.forEach((connector) => {
+      tasks.push(
+        this.connectorNormalisationService.normalise(connector, model)
+      );
+    });
+
+    model.normalizedConnectors = await Promise.all(tasks).then((x) => x.flat());
 
     return model;
   }
 
-  public async assignUris(model: DataRegistry, uriRegistry: UriRegistry): Promise<UriRegistry> {
+  public async assignUris(
+    model: DataRegistry,
+    uriRegistry: UriRegistry
+  ): Promise<UriRegistry> {
     uriRegistry.connectorOsloIdUriMap = new Map<number, URL>();
     const diagramConnectors: NormalizedConnector[] = [];
 
-    model.targetDiagram.connectorsIds.forEach(connectorId => {
-      const filteredConnectors = model.normalizedConnectors.filter(x => x.originalId === connectorId) || [];
+    model.targetDiagram.connectorsIds.forEach((connectorId) => {
```

Comment:
This can be simplified using `.flatMap`

```js
    const diagramConnectors: NormalizedConnector[] = model.targetDiagram.connectorsIds.flatMap(connectorId => {
      const filteredConnectors =
        model.normalizedConnectors.filter(
          x => x.originalId === connectorId,
        ) || [];
      return filteredConnectors;
    });
```